react-toggle-switch/my-app/src/ToggleSwitch.tsx

```diff
@@ -0,0 +1,21 @@
+import { useState } from 'react';
+import './App.css';
+
+export function ToggleSwitch() {
+  const [toggle, setToggle] = useState('off');
+
+  return (
+    <>
+      <div className="container">
+        <div className={toggle}>
+          <div
+            onClick={() => {
+              toggle === 'off' ? setToggle('on') : setToggle('off');
+            }}
+            className="switch"></div>
+        </div>
+        <div className="on-off-text">{toggle}</div>
```

Comment:
It's not a good idea to use a code state ('on' or 'off') to also display as text, because if you translate the product to a different language, it will stop working correctly. (Or maybe you just change your mind about the word or capitalization of the word.)